### PR TITLE
core/query: fix Item_field、type by MySQL8

### PR DESCRIPTION
**Pull request by DandreChen**
1.st_select_lex_unit -> Query_expression
	2.first_select() -> first_query_block()
	3.Item_equal_iterator is deleted
	4.Item_ref::DIRECT_REF is deleted
	5.escape -> escape()

### CODE REVIEW

File: storage/tianmu/core/query.cpp [hunk 1]
```diff
@@ -1106,19 +1106,22 @@ int Query::Item2CQTerm(Item *an_arg, CQTerm &term, const TabID &tmp_table, CondT
...
+  li = conds->get_fields().begin();
+  // stonedb8 end
+
   Item *const_item = conds->get_const();
   if (const_item) {
     if (!Item2CQTerm(const_item, zero_term, tmp_table, filter_type)) return CondID(-1);
   } else {
     ifield = li++;
-    if (!Item2CQTerm(ifield, zero_term, tmp_table, filter_type)) return CondID(-1);
+    if (!Item2CQTerm(&*ifield, zero_term, tmp_table, filter_type)) return CondID(-1); // stonedb8
```
Comment (andyli029) [inline storage/tianmu/core/query.cpp:1121]: add TODO

Comment (DandreChen) [reply storage/tianmu/core/query.cpp:1121]: #58

**Comment by DandreChen** [comment]
#58

**Review by andyli029 — APPROVED** [review]
LGTM

**Pull request merged**